docs/utilities/idempotency.md

```diff
@@ -77,31 +77,68 @@ If you're not [changing the default configuration for the DynamoDB persistence l
 
-```yaml hl_lines="5-13 21-23" title="AWS Serverless Application Model (SAM) example"
-Resources:
-  IdempotencyTable:
-	Type: AWS::DynamoDB::Table
-	Properties:
-	  AttributeDefinitions:
-		-   AttributeName: id
-			AttributeType: S
-	  KeySchema:
-		-   AttributeName: id
-			KeyType: HASH
-	  TimeToLiveSpecification:
-		AttributeName: expiration
-		Enabled: true
-	  BillingMode: PAY_PER_REQUEST
-
-  HelloWorldFunction:
-  Type: AWS::Serverless::Function
-  Properties:
-	Runtime: python3.9
-	...
-	Policies:
-	  - DynamoDBCrudPolicy:
-		  TableName: !Ref IdempotencyTable
-```
+=== "sam.yaml"
+
+    ```yaml hl_lines="5-13 21-23" title="AWS Serverless Application Model (SAM) example"
+    Resources:
+    IdempotencyTable:
+        Type: AWS::DynamoDB::Table
+        Properties:
+        AttributeDefinitions:
+            -   AttributeName: id
+                AttributeType: S
+        KeySchema:
+            -   AttributeName: id
+                KeyType: HASH
+        TimeToLiveSpecification:
+            AttributeName: expiration
+            Enabled: true
+        BillingMode: PAY_PER_REQUEST
+
+    HelloWorldFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+        Runtime: python3.9
+        ...
+        Policies:
+        - DynamoDBCrudPolicy:
+            TableName: !Ref IdempotencyTable
+    ```
+=== "cdk.py"
+
+    ```python hl_lines="14 17 26-29"  title="AWS Cloud Development Kit (CDK) Construct example"
+    from aws_cdk import RemovalPolicy
+    from aws_cdk import aws_dynamodb as dynamodb
+    from aws_cdk import aws_iam as iam
+    from constructs import Construct
+
+
+    class IdempotencyConstruct(Construct):
+
+        def __init__(self, scope: Construct, id_: str, lambda_role: iam.Role) -> None:
+            super().__init__(scope, id_)
+            self.idempotency_table = dynamodb.Table(
+                self,
+                'IdempotencyTable',
+                partition_key=dynamodb.Attribute(name='id', type=dynamodb.AttributeType.STRING),
+                billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
+                removal_policy=RemovalPolicy.DESTROY,
+                time_to_live_attribute='expiration',
+                point_in_time_recovery=True,
+            )
+            lambda_role.attach_inline_policy(
+            iam.Policy(
+                self,
+                'idempotency-policy',
+                statements=[
+                    iam.PolicyStatement(
+                        actions=['dynamodb:PutItem', 'dynamodb:GetItem', 'dynamodb:UpdateItem', 'dynamodb:DeleteItem'],
+                        resources=[self.idempotency_table.table_arn],
+                        effect=iam.Effect.ALLOW,
+                    )
+                ]
+            ))
+    ```
```

Comment:
Can we move the SAM and CDK samples into `examples/` folder?

Context: we are taking every opportunity we get to move code snippets into examples/, where our linting and type checker can run, so we make sure the code is good to be copy/pasted.

Reply:
done